src/Framework/PaymentGateways/Contracts/PaymentGatewayInterface.php

```diff
@@ -7,11 +7,12 @@
 use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
 use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
 use Give\PaymentGateways\DataTransferObjects\GatewaySubscriptionData;
+use Give\Subscriptions\Models\Subscription;
 
 /**
  * @since 2.18.0
  */
-interface PaymentGatewayInterface
+interface PaymentGatewayInterface extends SubscriptionModuleInterface
```

Comment:
@ravinderk I realized that, ultimately, the gateway itself needs to do basically everything the module does. Perhaps we'll break this apart later, but for now folks using the gateway shouldn't be aware of the underlying module architecture, as it is possible. So I removed the subscription methods and extended the subscription interface.